docker/Dockerfile.prereq

```diff
@@ -52,6 +53,6 @@ RUN if [ ! -f "/build/llvm-project/build/CMakeCache.txt" ]; then \
     fi
 
 ENV MAKEFLAGS=-j2
-RUN if timeout 30m cmake --build . --target -- ${MAKEFLAGS} ; then \
-      cmake --build . --target check-mlir; \
+RUN if cmake --build . --target -- ${MAKEFLAGS} ; then \
```

Comment:
Hi @gongsu832 this timeout is important because Travis has a 60m time limit for each job. So to workaround this timelimit, I broke down the llvm-project compilation into several stages, each stage is roughly doing "make all" for a maximum of 30 min, and then it gets killed, the intermediate builds are sent to DockerHub, and the next Travis job picks up from where the previous one lefts off from DockerHub. After a few iterations, we will have a complete llvm/mlir build on DockerHub.

So I don't think we can omit the timeout.

Reply:
Good catch. Yes I see it's needed for the 5-stage build. I was playing around and thought I added it back but apparently forgot it. Thanks.